packages/__docs__/buildScripts/ai-accessible-documentation/generate-ai-accessible-markdowns.mts

```diff
@@ -0,0 +1,283 @@
+
+    propsContent += `| ${prop.component} | ${prop.name} | ${type} | ${required} | ${defaultValue} | ${description} |\n`
+  }
+  propsContent += '\n'
+
+  return propsContent
+}
+
+function generateComponentUsage(doc: DocumentationData): string {
+  if (!isComponent(doc)) return ''
+
+  const { id, displayName, packageName, esPath } = doc
+  const importName = displayName || id
+
+  const example = []
+
+  if (packageName) {
+    example.push(`/*** ES Modules (with tree shaking) ***/
+import { ${importName} } from '${packageName}'`)
+  }
+
+  if (esPath) {
+    example.push(`/*** ES Modules (without tree shaking) ***/
+import { ${importName} } from '${esPath}'`)
+  }
+
+  if (example.length === 0) return ''
+
+  let usageContent = `### Usage\n\n`
+
+  if (packageName) {
+    usageContent += `Install the package:\n\n\`\`\`shell\nnpm install ${packageName}\n\`\`\`\n\n`
+  }
+
+  usageContent += `Import the component:\n\n\`\`\`javascript\n${example.join(
+    '\n\n'
+  )}\n\`\`\`\n\n`
+
+  return usageContent
+}
+
+function generateComponentMarkdown(
+  jsonData: ComponentData,
+  childComponents: ComponentData[] = []
+): string {
+  const { displayName, description } = jsonData
+
+  let markdownContent = `# ${displayName}\n\n`
+  markdownContent += `${description}\n\n`
+  markdownContent += generatePropsTable(jsonData, childComponents)
+  markdownContent += generateComponentUsage(jsonData)
+
+  return markdownContent
+}
+
+function generateGuideMarkdown(jsonData: GuideData): string {
+  const { description } = jsonData
+
+  let markdownContent = `${description}\n\n`
+
+  return markdownContent
+}
+
+async function generateAIAccessibleMarkdowns(
+  docsFolder: string,
+  outputDir: string
+): Promise<void> {
+  if (!existsSync(outputDir)) {
+    mkdirSync(outputDir, { recursive: true })
+    console.log(`Created markdown directory: ${outputDir}`)
+  }
+
+  const files = readdirSync(docsFolder)
+  const jsonFiles = files.filter((file) => file.endsWith('.json'))
+
+  const allDocs: DocumentationData[] = []
+  for (const file of jsonFiles) {
+    try {
+      const filePath = path.join(docsFolder, file)
+      const fileContent = readFileSync(filePath, 'utf-8')
+      const jsonData: DocumentationData = JSON.parse(fileContent)
+      allDocs.push(jsonData)
+    } catch (error) {
+      console.error(`Error processing file ${file}:`, error)
+    }
+  }
+
+  // Process guides
+  const guides = allDocs.filter(isGuide)
+  for (const guide of guides) {
+    const markdownContent = generateGuideMarkdown(guide)
+    const fileName = `${
+      guide.id || guide.title.toLowerCase().replace(/\s+/g, '-')
+    }.md`
+    const filePath = path.join(outputDir, fileName)
+    writeFileSync(filePath, markdownContent)
+  }
+
+  // Process components
+  const components = allDocs.filter(isComponent)
+  for (const component of components) {
+    const childComponents = components.filter(
+      (c) => c.parent === component.displayName
+    )
```

Comment:
I think this is not working. The generated files for `Tabs` or `Table` do not list any child components.

Reply:
looks good now!